modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/endpoint/EndpointMatcher.java

```diff
@@ -72,10 +72,7 @@ private String _getSingleElementRegex(APIApplication.Endpoint endpoint) {
 
 		endpointPath = endpointPath + StringPool.FORWARD_SLASH;
 
-		if (Objects.equals(
-				endpoint.getPathParameter(),
-				APIApplication.Endpoint.PathParameter.ID)) {
-
+		if (Objects.equals(endpoint.getPathParameter(), "id")) {
```

Comment:
I am not sure if we need this as any other long parameter whose name is not id will be validated against the `\\S+` regex